index.js

```diff
@@ -1,2 +1,601 @@
+ * @param {String} options.token Passcode to validate
+ * @param {Integer} options.counter Counter value. This should be stored by
+ *   the application and must be incremented for each request.
+ * @param {Integer} [options.digits=6] The number of digits for the one-time
+ *   passcode.
+ * @param {Integer} [options.window=0] The allowable margin for the counter.
+ *   The function will check "W" codes in the future against the provided
+ *   passcode, e.g. if W = 10, and C = 5, this function will check the
+ *   passcode against all One Time Passcodes between 5 and 15, inclusive.
+ * @param {String} [options.encoding="ascii"] Key encoding (ascii, hex,
+ *   base32, base64).
+ * @param {String} [options.algorithm="sha1"] Hash algorithm (sha1, sha256,
+ *   sha512).
+ * @return {Object} On success, returns an object with the counter
+ *   difference between the client and the server as the `delta` property (i.e.
+ *   `{ delta: 0 }`).
+ * @method hotp․verifyDelta
+ * @global
+ */
+
+exports.hotp.verifyDelta = function hotpVerifyDelta (options) {
+  var i;
+
+  // shadow options
+  options = Object.create(options);
+
+  // unpack options
+  var token = options.token;
+  var window = parseInt(options.window || 0, 10);
+  var counter = parseInt(options.counter || 0, 10);
+
+  // loop from C to C + W
+  for (i = counter; i <= counter + window; ++i) {
+    options.counter = i;
+    if (exports.hotp(options) === token) {
+      // found a matching code, return delta
+      return {delta: i - counter};
+    }
+  }
+
+// no codes have matched
+};
+
+/**
+ * Verify a counter-based one-time token against the secret and return true if
+ * it verifies. Helper function for `hotp.verifyDelta()`` that returns a boolean
+ * instead of an object. For more on how to use a window with this, see
+ * {@link hotp.verifyDelta}.
+ *
+ * @param {Object} options
+ * @param {String} options.secret Shared secret key
+ * @param {String} options.token Passcode to validate
+ * @param {Integer} options.counter Counter value. This should be stored by
+ *   the application and must be incremented for each request.
+ * @param {Integer} [options.digits=6] The number of digits for the one-time
+ *   passcode.
+ * @param {Integer} [options.window=0] The allowable margin for the counter.
+ *   The function will check "W" codes in the future against the provided
+ *   passcode, e.g. if W = 10, and C = 5, this function will check the
+ *   passcode against all One Time Passcodes between 5 and 15, inclusive.
+ * @param {String} [options.encoding="ascii"] Key encoding (ascii, hex,
+ *   base32, base64).
+ * @param {String} [options.algorithm="sha1"] Hash algorithm (sha1, sha256,
+ *   sha512).
+ * @return {Boolean} Returns true if the token matches within the given
+ *   window, false otherwise.
+ * @method hotp․verify
+ * @global
+ */
+exports.hotp.verify = function hotpVerify (options) {
+  return exports.hotp.verifyDelta(options) != null;
```

Comment:
Actually, I'm wrong. I remember it complained because of an equality with `null` (see below) but it seems to only complain with `x == null`, not `x != null`. Sorry about that.

https://github.com/speakeasyjs/speakeasy/commit/a8c306ff14018cfd39ac59779ef5eadcd57c5213#diff-168726dbe96b3ce427e7fedce31bb0bcL551